FlowCrypt/Functionality/Mail Provider/MessagesList Provider/Gmail+MessagesList.swift

```diff
@@ -8,28 +8,23 @@
 
 import FlowCryptCommon
 import GoogleAPIClientForREST_Gmail
-import Promises
 
 extension GmailService: MessagesListProvider {
-    func fetchMessages(using context: FetchMessageContext) -> Promise<MessageContext> {
-        Promise { resolve, reject in
-            let list = try awaitPromise(fetchMessagesList(using: context))
-            let messageRequests: [Promise<Message>] = list.messages?.compactMap(\.identifier).map(fetchFullMessage(with:)) ?? []
-
-            all(messageRequests)
-                .then { messages in
-                    let context = MessageContext(messages: messages, pagination: .byNextPage(token: list.nextPageToken))
-                    resolve(context)
-                }
-                .catch { error in
-                    reject(error)
-                }
+    func fetchMessages(using context: FetchMessageContext) async throws -> MessageContext {
+        let list = try await fetchMessagesList(using: context)
+        let messageIdentifiers = list.messages?.compactMap(\.identifier) ?? []
+
+        var messages: [Message] = []
+        for identifier in messageIdentifiers {
+            messages.append(try await fetchFullMessage(with: identifier))
         }
```

Comment:
In a separate issue, should make sure these pull in parallel instead of sequentially.